src/jit/emitxarch.cpp

```diff
@@ -4812,6 +4813,22 @@ void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1,
         emitIns_R_R(ins, emitTypeSize(simdtype), reg, reg2);
     }
 }
+
+void emitter::emitIns_SIMD_R_R_R_I(instruction ins, regNumber reg, regNumber reg1, regNumber reg2, int ival, var_types simdtype)
+{
+    if (UseVEXEncoding() && reg1 != reg)
```

Comment:
For both this and `emitIns_SIMD_R_R_R` above: Do we have a work item tracking support for `reg2` being a memory operand when it is properly aligned?

Currently we are generating things like
```
C4E1791000           vmovupd  xmm0, xmmword ptr [rax]
C4E14858F0           vaddps   xmm6, xmm0
```

While in some cases we could generate `vaddps xmm6, [rax]`

Reply:
> reg2 being a memory operand when it is properly aligned?

And in what circumstances will the memory operand be aligned? Only for local variables I guess.

Reply:
Yeah. It definitely can't be used everywhere, and isn't something I think needs to be immediate.

But, for the cases where the memory will always be aligned (locals, is one example), it would be useful to generate the more efficient code

Reply:
> Only for local variables I guess

@tannergooding @mikedn
 
It is not true. In general devs writing vectorized code would peel off start of loops to align data and do all the remaining calculations on aligned data until the end of the loops where processing may require change of logic.

This means  that all data accessed in the `main` loops will be aligned and doing anything but `vaddps xmm6, [rax]` would be an unacceptable loss of the CPU cycles.

Reply:
> In general devs writing vectorized code would peel off start of loops to align data

The key here is "in general". The JIT has no way to know that the address is aligned or not and it cannot make such assumptions simply because "in general" developers do that.

I suspect that it will work if the developer uses `LoadAligned` to indicate that the memory address is aligned.

Reply:
I agree that, "in general", users writing performant code will try to keep accesses aligned to avoid reads/writes that cross cache line or page boundaries.

However, as @mikedn points out, assuming this is incorrect and detecting it is likely hard.

It's probably worth a separate discussion (will log a bug in a bit) but I believe it is something we should try to enable (maybe a special intrinsic, compiler hint,  or attribute to tell the jit memory accesses will be aligned for a given method).

Reply:
> It's probably worth a separate discussion (will log a bug in a bit) 

It is the right approach I think as this not that easy to get it right